src/internal/eval-tidy.c

```diff
@@ -379,8 +377,17 @@ sexp* rlang_eval_tidy(sexp* expr, sexp* data, sexp* frame) {
 const char* data_pronoun_c_names[4] = { "src", "lookup_msg", "read_only", NULL };
 
 void rlang_init_eval_tidy() {
-  tilde_thunk_fmls = rlang_constants_get("tilde_thunk_fmls");
-  tilde_thunk_body = rlang_constants_get("tilde_thunk_body");
+  tilde_fn = r_parse_eval(
```

Comment:
Is there a reason these have to be created in C code? Why can't we store in the package?

Reply:
It's completely equivalent but I think storing on the R side makes things a bit more complicated overall, especially as you start using this pattern more generally. It is nice to see the code inline.

Where it makes sense, I think we should replace the dynamic creation of language objects with cons/lang by parsing at load-time. It is faster, less error prone, and more flexible. In C this has to be done in an init function and stored in a global variable but in C++ you can dynamically initialise in a static-local variable, which is even nicer.

Too bad C / C++ don't have multiline literal strings though.

Reply:
But we optimise for reading the code? In this case I don't want to create these objects at load-time particularly. But being able to do it at load-time means that this is a general solution that even works in header-only libraries that don't have a namespace (esp. with C++).

I think parsing also has advantages when writing the code. It's easy to parse inline small pieces of code and when performance matters you move it to a static variable. Here's another example: https://github.com/r-lib/rlang/pull/601/commits/760f6fce2a5e73420ea0a920b7c7d49948beed19

You may be right that in this case it's better to build them in the namespace. It obviously depends on the size of the function, whether you have your own namespace, etc. As all programming there are tradeoffs, I'm just experimenting with this new idiom that I like so I can develop a feel for it.